src/shared/lib/StoreUtils.ts

```diff
@@ -1622,6 +1623,113 @@ export function makeGetOncoKbCnaAnnotationForOncoprint(
     }
 }
 
+export function getSampleClinicalDataMapByThreshold(
+    clinicalData: ClinicalData[],
+    clinicalAttributeId: string,
+    threshold: number
+) {
+    return _.reduce(
+        clinicalData,
+        (acc, next) => {
+            if (next.clinicalAttributeId === clinicalAttributeId) {
+                const value = getNumberClinicalDataValue(next);
+                if (value && value >= threshold) {
+                    acc[next.sampleId] = next;
+                }
+            }
+            return acc;
+        },
+        {} as { [key: string]: ClinicalData }
+    );
+}
+
+export function getNumberClinicalDataValue(
```

Comment:
Should we use `numerical` instead of `number` in the function name?
```suggestion
export function getNumericalClinicalDataValue(
```